spinalcordtoolbox/deepseg/inference.py

```diff
@@ -77,3 +91,158 @@ def segment_and_average_volumes(model_paths, input_filenames, options):
     target_lst = target_lsts[0]
 
     return nii_lst, target_lst
+
+
+def segment_non_ivadomed(path_model, model_type, input_filenames, threshold):
+    # MONAI and NNUnet have similar structure, and so we use nnunet+inference functions with the same signature
+    if model_type == "monai":
+        create_net = ds_monai.create_nnunet_from_plans
+        inference = segment_monai
+    else:
+        assert model_type == "nnunet"
+        create_net = ds_nnunet.create_nnunet_from_plans
+        inference = segment_nnunet
+
+    # equivalent to `with torch.no_grad()`
+    torch.set_grad_enabled(False)
+
+    # load model from checkpoint
+    net = create_net(path_model)
+
+    nii_lst, target_lst = [], []
+    for fname_in in input_filenames:
+        # model may be multiclass, so the `inference` func should output a list of fnames and targets
+        fnames_out, targets = inference(path_img=fname_in, tmpdir=tmp_create(basename="sct_deepseg"), predictor=net)
+        for fname_out, target in zip(fnames_out, targets):
+            # TODO: Use API binarization function when output filetype is sct.image.Image
+            run_proc(["sct_maths", "-i", fname_out, "-bin", str(threshold), "-o", fname_out])
+            # TODO: Change the output filetype from Nifti1Image to sct.image.Image to mitigate #3232
+            nii_lst.append(nib.load(fname_out))
```

Comment:
Right now, I'm using a clunky call to `sct_maths` to binarize the output images as per the existing `-thr` argument of `sct_deepseg`.

This could be done more elegantly if the inference functions output `spinalcordtoolbox.image.Image` objects, but unfortunately `sct_deepseg` currently expects `Nifti1Image` objects instead (see `nii_lst`).

Thankfully, I was actually already in the process of changing this logic separately because of https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3232#issuecomment-1885717752. So, what I think I might do is create a separate pull request to fix the float/int scaling issue. Then, when that PR is merged, I can update this PR to no longer use `Nifti1Images` as output.

Reply:
Resolved by https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4345/commits/337eae1d96ba9c3d29684a82079b7b4f32fd9d66! :)